src/molecular-weight.jl

```diff
@@ -0,0 +1,260 @@
+function molecular_weight(aa_seq::AASeq, unit=:"gmol")
+    AA_vector = Vector{Float64}(undef, 0)
+    i = 1
+    while i <= length(aa_seq)
+        if aa_seq[i] == AA_A
+            push!(AA_vector, AA_A_MW)
+        elseif aa_seq[i] == AA_R
+            push!(AA_vector, AA_R_MW)
+        elseif aa_seq[i] == AA_N
+            push!(AA_vector, AA_N_MW)
+        elseif aa_seq[i] == AA_D
+            push!(AA_vector, AA_D_MW)
+        elseif aa_seq[i] == AA_C
+            push!(AA_vector, AA_C_MW)
+        elseif aa_seq[i] == AA_Q
+            push!(AA_vector, AA_Q_MW)
+        elseif aa_seq[i] == AA_E
+            push!(AA_vector, AA_E_MW)
+        elseif aa_seq[i] == AA_G
+            push!(AA_vector, AA_G_MW)
+        elseif aa_seq[i] == AA_H
+            push!(AA_vector, AA_H_MW)  
+        elseif aa_seq[i] == AA_I
+            push!(AA_vector, AA_I_MW)
+        elseif aa_seq[i] == AA_L
+            push!(AA_vector, AA_L_MW)    
+        elseif aa_seq[i] == AA_K
+            push!(AA_vector, AA_K_MW)
+        elseif aa_seq[i] == AA_M
+            push!(AA_vector, AA_M_MW)                                            
+        elseif aa_seq[i] == AA_F
+            push!(AA_vector, AA_F_MW)
+        elseif aa_seq[i] == AA_P
+            push!(AA_vector, AA_P_MW)
+        elseif aa_seq[i] == AA_S
+            push!(AA_vector, AA_S_MW)
+        elseif aa_seq[i] == AA_T
+            push!(AA_vector, AA_T_MW)   
+        elseif aa_seq[i] == AA_W
+            push!(AA_vector, AA_W_MW)
+        elseif aa_seq[i] == AA_Y
+            push!(AA_vector, AA_Y_MW)
+        elseif aa_seq[i] == AA_V
+            push!(AA_vector, AA_V_MW)
+        elseif aa_seq[i] == AA_O
+            push!(AA_vector, AA_O_MW_MW)
+        elseif aa_seq[i] == AA_U
+            push!(AA_vector, AA_U_MW_MW)
+        elseif aa_seq[i] == AA_B
+            push!(AA_vector, AA_B_MW_MW)
+        elseif aa_seq[i] == AA_J
+            push!(AA_vector, AA_J_MW_MW)
+        elseif aa_seq[i] == AA_Z
+            push!(AA_vector, AA_Z_MW_MW)
+        elseif aa_seq[i] == AA_X
+            push!(AA_vector, AA_X_MW_MW)
+        elseif aa_seq[i] == AA_O_Gap
+            push!(AA_vector, AA_O_Gap_MW)                           
+        else
+            println("Error due to unknown amino acid at $i position")
+            return nothing
+        end
+        i = i+1
+    end
+        if unit == "gmol"
+            return round(sum(AA_vector) - ((length(aa_seq) - 1) * WATER_MW); digits = 3)
+        elseif unit == "kDa"
+            return round(((sum(AA_vector) - ((length(aa_seq) - 1) * WATER_MW))/1000); digits = 3)
+        else 
+            return nothing
+        end
+end
+
+# Defining a function, which accepts a AA sequence and print the molecular weight in g/mol and kDa from a given n_start position in the sequence until a n_end given position in the sequence
+function molecular_weight(aa_seq::AASeq, n_start::Integer, n_end::Integer)
+    AA_vector = Vector{Float64}(undef, 0)
```

Comment:
Delete this function. If you want to get the weight of a sequence, from, say, positions 5 to 10, you can use
```julia
julia> molecular_weight(view(aa_seq, 5:10))
```
This is efficient, and it's nicer for us to only have to maintain one function (also nicer for the user that there are less methods to reason about)